robot.py

```diff
@@ -147,13 +161,9 @@ def turn(self, speed, deg, port):
                     if not self.__gyro.angle() - startValue > deg * 0.6:
                         speed = speed - _map(deg, 1, 360, 10, 0.1) if speed > 20 else speed
 
-                    #cancel if button pressed
-                    if any(self.brick.buttons()):
```

Comment:
this is needed in here to exit the while loop
if you put that outside the loop, it will not stop the current action, but after the angle is fully turned.
Since the button is meant to stop all actions immediately this would not be the desired behaviour

Reply:
same goes btw. for the two other times below...